java/src/jmri/jmrit/logixng/Base.java

```diff
@@ -485,7 +485,11 @@ public void printTree(
             PrintWriter writer,
             String indent,
             String currentIndent);
-    
+
+    public void getUsageTree(int level, NamedBean bean, List<jmri.NamedBeanUsageReport> report, NamedBean cdl);
+
+    public void getUsageDetail(int level, NamedBean bean, List<jmri.NamedBeanUsageReport> report, NamedBean cdl);
```

Comment:
It would be good to have a Javadoc on these methods:
* What is `level`?
* I assume that `bean` is the bean that is searched for. Can `bean` be null, and if so, will that give me a list of all beans that is used?
* What is `cdl`?

Reply:
`level` was used by the debug (info) statements to make sure that I was handling the recursion in the tree properly.

When `getUsageReport` is invoked using the Where Used tool it will never be null.  When invoked from another location, such as a script, it might be null.  If it is null (or nothing found), an empty report list will be returned.

`cdl` is the name of the current conditional.  It is passed along for use when the NamedBeanUsageReport is created.

Reply:
Thanks. It would be good if you could add this as Javadoc to these methods. It makes it easier to implement these methods in other actions and expressions, for example ActionTurnout and ExpressionSensor.

Reply:
Only `getUsageDetail` is needed for the actions and expressions.  The PR includes Blocks and Sensors.

There is a slight variation between actions and expressions.  There is a keyword that is used by `WhereUsedCollectors`: **_ConditionalNGAction_** or **_ConditionalNGExpression_**.

My convention is that the usage related methods are placed at the end of the class.

Reply:
> Only `getUsageDetail` is needed for the actions and expressions. The PR includes Blocks and Sensors.
> 
> There is a slight variation between actions and expressions. There is a keyword that is used by `WhereUsedCollectors`: **_ConditionalNGAction_** or **_ConditionalNGExpression_**.
> 
> My convention is that the usage related methods are placed at the end of the class.

I'm fine with that. But it would be good if you could add all this info too to the Javadoc. Later, when I or somebody else adds this feature to a new action/expression, we will look at the javadoc and not at this PR.

Thanks.